frappe_s3_attachment/controller.py

```diff
@@ -292,6 +305,91 @@ def upload_existing_files_s3(name, file_name):
     else:
         pass
 
+# download s3 files
+def download_s3_files(private_local_folder_path, public_local_folder_path):
+    s3_download = S3Operations()
+    bucket = s3_download.S3_RESOURCE.Bucket(s3_download.BUCKET)
+    for obj in bucket.objects.filter(Prefix=s3_download.folder_name):
+        # Check if the object is a file and not a folder
+        if not obj.key.endswith('/'):
+            download_s3_file(obj, s3_download.BUCKET, private_local_folder_path, public_local_folder_path)
+
+# download s3 file
+def download_s3_file(obj, bucket_name, private_local_folder_path, public_local_folder_path):
+    # check from file_url that file is in s3 or local and ommit download if it is in local
+    name = frappe.db.sql(f"""select `name` from `tabFile` where `file_url` LIKE '%{obj.key}%'""")
+    if len(name)==0:
+        return
+    
+    acl = obj.Acl()
+    
+    s3 = S3Operations()
+    s3_object = s3.S3_RESOURCE.Object(str(bucket_name), str(obj.key))
+    s3FileName = obj.key.split('/')[-1]
+    fileName = s3FileName.split('_',1)[1]
+    max_retries = 5
+    retry_delay = 0.5
+
+    if len(acl.grants) != 1:
+        if acl.grants[1]['Grantee']['Type'] == 'Group' and acl.grants[1]['Grantee']['URI'] == 'http://acs.amazonaws.com/groups/global/AllUsers':
```

Comment:
Shouldn't this be stored in the Files doctype? Even if it's not, I think we should use the /public path prefix instead of ACL to determine the file's original location.

Reply:
Ok